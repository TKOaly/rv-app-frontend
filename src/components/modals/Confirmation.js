import './styles/Confirmation.scss';
import { closeModal } from '../../reducers/modalReducer';
import { connect } from 'react-redux';
import { increaseBalance } from '../../reducers/userReducer';
import { resetAmount, toggleConfirmationVisibility } from '../../reducers/depositReducer';
import React from 'react';
import moneyFormatter from '../../services/moneyFormatter';

export class Confirmation extends React.Component {
    componentDidMount = () => {
        this.oldFocusedElement = document.activeElement;
        this.dialog.focus();
    };

    handleConfirm = (event) => {
        event.preventDefault();
        this.props.increaseBalance(this.props.token, this.props.depositAmount);
        this.props.closeModal();
    };

    handleCancel = (event) => {
        event.preventDefault();
        this.props.toggleConfirmationVisibility(false);
        if (this.oldFocusedElement) {
            this.oldFocusedElement.focus();
        }
    };

    handleKeyDown = (event) => {
        if (event.key === 'Enter') {
            event.preventDefault();
            this.handleConfirm(event);
        } else if (event.key === 'Escape') {
            event.preventDefault();
            this.handleCancel(event);
        }
    };

    dialogRef = (dialog) => {
        this.dialog = dialog;
    };

    render = () => {
        const { depositAmount } = this.props;
        return (
            <div className="confirm-overlay">
                <div className="confirm" tabIndex="0" ref={this.dialogRef} onKeyDown={this.handleKeyDown}>
                    <span className="confirm-title-text">Confirm <b>{moneyFormatter.centsToString(depositAmount)} â‚¬</b> deposit</span>
                    <br />
                    <br />
                    <button className="btn number cancel cancelBtn confirmation-cancel-btn" onClick={this.handleCancel}>
                        Cancel (ESC)
                    </button>
                    <button className="btn number success confirmBtn confirmation-ok-btn" onClick={this.handleConfirm}>
                        OK (ENTER)
                    </button>
                </div>
            </div>
        );
    };
}

const mapStateToProps = (state) => {
    return {
        token: state.authentication.access_token
    };
};

const mapDispatchToProps = {
    increaseBalance,
    resetAmount,
    closeModal,
    toggleConfirmationVisibility
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Confirmation);
