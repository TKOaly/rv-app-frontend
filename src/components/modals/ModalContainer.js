import './styles/ModalContainer.css';
import { CSSTransition } from 'react-transition-group';
import { closeModal } from '../../reducers/modalReducer';
import { connect } from 'react-redux';
import React from 'react';

export class ModalContainer extends React.Component {
    handleClose = (e) => {
        this.props.closeModal();
    };

    handleModalClick = (e) => e.stopPropagation();

    render = () => {
        const ModalContent = this.props.contentComponent;
        if (!ModalContent) {
            return null;
        }

        return (
            <CSSTransition in={this.props.modalVisible} timeout={200} classNames="overlay" mountOnEnter unmountOnExit>
                {(state) => {
                    return (
                        <div className="modal-overlay" onClick={this.handleClose}>
                            <div className="modal-container" onClick={this.handleModalClick}>
                                <ModalContent {...this.props.modalProps} />
                            </div>
                        </div>
                    );
                }}
            </CSSTransition>
        );
    };
}

const mapDispatchToProps = {
    closeModal
};

const mapStateToProps = (state) => ({
    contentComponent: state.modal.modalContent,
    modalProps: state.modal.props,
    modalVisible: state.modal.modalVisible
});

export default connect(mapStateToProps, mapDispatchToProps)(ModalContainer);
