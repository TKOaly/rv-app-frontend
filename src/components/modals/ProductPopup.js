import './styles/ProductPopup.css';
import { buyProduct, setBuyAmount } from '../../reducers/productReducer';
import { closeModal } from '../../reducers/modalReducer';
import { connect } from 'react-redux';
import React from 'react';
import moneyFormatter from '../../services/moneyFormatter';

class ProductPopup extends React.Component {
    handleBuy = () => {
        this.props.buyProduct(this.props.product, this.props.buyAmount);
        this.props.closeModal();
    };

    handleCancel = () => {
        this.props.closeModal();
    };

    handleDecreaseAmount = () => {
        if (this.props.buyAmount > 1) {
            this.props.setBuyAmount(this.props.buyAmount - 1);
        }
    };

    handleIncreaseAmount = () => {
        this.props.setBuyAmount(this.props.buyAmount + 1);
    };

    componentWillUnmount = () => {
        // Reset amount back to 1
        this.props.setBuyAmount(1);
    };

    render = () => {
        return (
            <div className="product-popup">
                <div className="product-popup-header">
                    <h3>Buy product</h3>
                </div>
                <div className="product-info">
                    <div className="product-name">{this.props.product.product_name}</div>
                </div>
                <div className="product-quantity">
                    <div>Amount:</div>
                    <div className="quantity-picker">
                        <button onClick={this.handleDecreaseAmount}>-</button>
                        <span>{this.props.buyAmount} pcs</span>
                        <button onClick={this.handleIncreaseAmount}>+</button>
                    </div>
                    <div className="total">
                        <div>Total</div>
                        <div className="sum">
                            {moneyFormatter.centsToString(this.props.product.sellprice * this.props.buyAmount)} â‚¬
                        </div>
                    </div>
                </div>
                <div className="product-actions">
                    <button className="cancel-btn" onClick={this.handleCancel}>
                        Cancel
                    </button>
                    <button className="purchase-btn" onClick={this.handleBuy}>
                        Buy
                    </button>
                </div>
            </div>
        );
    };
}

const mapStateToProps = (state) => {
    return {
        buyAmount: state.products.buyAmount
    };
};

const mapDispatchToProps = {
    closeModal,
    buyProduct,
    setBuyAmount
};

export default connect(mapStateToProps, mapDispatchToProps)(ProductPopup);
