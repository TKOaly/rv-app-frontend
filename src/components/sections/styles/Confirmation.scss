@import "vars";

.confirm-overlay {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
}

.confirm {
  position: absolute;
  padding-top: 20px;
  padding-bottom: 20px;
  box-sizing: border-box;
  -webkit-box-sizing: border-box;
  width: 70%;
  z-index: 20;
  color: black;
  text-align: center;
  font-size: 20pt;
  left: calc((100% - 70%) / 2);
  top: calc((100% - 70%) / 2);
  background-color: white;
  box-shadow: $box-shadow
}
