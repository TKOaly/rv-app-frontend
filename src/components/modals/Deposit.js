import './styles/Deposit.scss';
import { Fade } from '../animations/Animations';
import { TransitionGroup } from 'react-transition-group';
import { closeModal } from '../../reducers/modalReducer';
import { connect } from 'react-redux';
import { errorMessage } from '../../reducers/notificationReducer';
import {
    resetAmount,
    resetDeposit,
    setAmountText,
    toggleConfirmationVisibility
} from '../../reducers/depositReducer';
import Confirmation from './Confirmation';
import React from 'react';
import moneyFormatter from '../../services/moneyFormatter';

// TODO: Add own styles
export const OkButton = ({ handleSubmit }) => (
    <button className="btn number success" onClick={handleSubmit}>
    OK (ENTER)
    </button>
);

// TODO: Add own styles
export const CancelButton = ({ handleCancel }) => (
    <button className="btn number cancel" onClick={handleCancel}>
    Cancel (ESC)
    </button>
);

export class Deposit extends React.Component {
  static depositAmountTextToCents = (text) => {
      if (text === '') {
          return 0;
      } else {
          return moneyFormatter.stringToCents(text);
      }
  };

  componentDidMount = () => {
      this.oldFocusedElement = document.activeElement;
      this.textField.focus();
  };

  componentWillUnmount = () => {
      this.props.resetDeposit();
      if (this.oldFocusedElement) {
          this.oldFocusedElement.focus();
      }
  };

  createIncrementHandler = (increment) => {
      return (event) => {
          event.preventDefault();
          const cents = Deposit.depositAmountTextToCents(
              this.props.depositAmountText
          );
          if (Number.isNaN(cents)) {
              this.props.errorMessage(
                  'Invalid deposit amount in text field. Use format 10.00'
              );
          } else {
              this.props.setAmountText(
                  moneyFormatter.centsToString(cents + increment)
              );
          }
      };
  };

  handleSubmit = () => {
      const cents = Deposit.depositAmountTextToCents(
          this.props.depositAmountText
      );
      if (Number.isNaN(cents)) {
          this.props.errorMessage(
              'Invalid deposit amount in text field. Use format 10.00'
          );
      } else if (cents <= 0) {
          this.props.errorMessage('You cannot deposit 0 € or less.');
      } else {
          this.props.toggleConfirmationVisibility(true);
      }
  };

  handleCancel = () => {
      this.props.closeModal();
  };

  handleKeyDown = (event) => {
      if (event.key === 'Enter') {
          event.preventDefault();
          this.handleSubmit();
      } else if (event.key === 'Escape') {
          event.preventDefault();
          this.handleCancel();
      }
  };

  handleValueChange = (event) => {
      this.props.setAmountText(event.target.value);
  };

  depositRef = (input) => {
      this.textField = input;
  };

  render = () => {
      return (
          <div className="deposit-wrapper">
              <div className="deposit">
                  <div className="input-wrapper">
                      <input
                          className="input"
                          type="text"
                          placeholder="0.00"
                          value={this.props.depositAmountText}
                          onKeyDown={this.handleKeyDown}
                          onChange={this.handleValueChange}
                          ref={this.depositRef}
                      />
                      <div className="euro-sign">€</div>
                  </div>
                  <button className="btn number erase" onClick={this.props.resetAmount}>
            Reset
                  </button>
                  <button
                      className="btn number increment"
                      onClick={this.createIncrementHandler(5)}
                  >
            + 0.05 €
                  </button>
                  <button
                      className="btn number increment"
                      onClick={this.createIncrementHandler(10)}
                  >
            + 0.10 €
                  </button>
                  <button
                      className="btn number increment"
                      onClick={this.createIncrementHandler(20)}
                  >
            + 0.20 €
                  </button>
                  <button
                      className="btn number increment"
                      onClick={this.createIncrementHandler(50)}
                  >
            + 0.50 €
                  </button>
                  <button
                      className="btn number increment"
                      onClick={this.createIncrementHandler(100)}
                  >
            + 1.00 €
                  </button>
                  <button
                      className="btn number increment"
                      onClick={this.createIncrementHandler(200)}
                  >
            + 2.00 €
                  </button>
                  <button
                      className="btn number increment"
                      onClick={this.createIncrementHandler(500)}
                  >
            + 5.00 €
                  </button>
                  <button
                      className="btn number increment"
                      onClick={this.createIncrementHandler(1000)}
                  >
            + 10.00 €
                  </button>
                  <button
                      className="btn number increment"
                      onClick={this.createIncrementHandler(2000)}
                  >
            + 20.00 €
                  </button>
                  <CancelButton handleCancel={this.handleCancel} />
                  <button
                      className="btn number increment"
                      onClick={this.createIncrementHandler(5000)}
                  >
            + 50.00 €
                  </button>
                  <OkButton handleSubmit={this.handleSubmit} />
              </div>
              <TransitionGroup>
                  {this.props.confirmationVisibility && (
                      <Fade>
                          <Confirmation
                              depositAmount={Deposit.depositAmountTextToCents(
                                  this.props.depositAmountText
                              )}
                          />
                      </Fade>
                  )}
              </TransitionGroup>
          </div>
      );
  };
}

const mapStateToProps = (state) => {
    return {
        depositAmountText: state.deposit.depositAmountText,
        confirmationVisibility: state.deposit.confirmationVisibility
    };
};

const mapDispatchToProps = {
    setAmountText,
    resetAmount,
    resetDeposit,
    toggleConfirmationVisibility,
    errorMessage,
    closeModal
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Deposit);
