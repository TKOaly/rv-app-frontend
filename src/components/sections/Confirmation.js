import './styles/Confirmation.css';
import React from 'react';
import moneyFormatter from '../../services/moneyFormatter';

class Confirmation extends React.Component {
    handleConfirm = (event) => {
        this.props.increaseBalance(this.props.token, this.props.depositAmount);
        this.props.resetAmount();
        this.props.toggleConfirmationVisibility(false);
        this.props.closeModal();
    };

    handleCancel = (event) => {
        this.props.toggleConfirmationVisibility(false);
    };

    render = () => {
        return (
            <div className="confirm-overlay">
                <div className="confirm">
                    Confirm <b>{moneyFormatter.centsToString(this.props.depositAmount)} â‚¬</b> deposit
                    <br />
                    <br />
                    <button className="btn number cancel cancelBtn" onClick={this.handleCancel}>
                        Cancel
                    </button>
                    <button className="btn number success confirmBtn" onClick={this.handleConfirm}>
                        OK
                    </button>
                </div>
            </div>
        );
    };
}

export default Confirmation;
