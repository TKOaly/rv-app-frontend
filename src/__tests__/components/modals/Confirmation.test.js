import { Confirmation } from '../../../components/modals/Confirmation';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<Confirmation />', () => {
    it('should match snapshot', () => {
        const cmpnt = renderer
            .create(<Confirmation />, {
                createNodeMock: () => {
                    return {
                        focus: () => {}
                    };
                }
            })
            .toJSON();
        expect(cmpnt).toMatchSnapshot();
    });
});

