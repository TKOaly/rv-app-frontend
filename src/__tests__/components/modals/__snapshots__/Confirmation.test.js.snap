// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Confirmation /> should match snapshot 1`] = `
<div
  className="confirm-overlay"
>
  <div
    className="confirm"
    onKeyDown={[Function]}
    tabIndex="0"
  >
    <span
      className="confirm-title-text"
    >
      Confirm 
      <b>
        NaN.NaN
         â‚¬
      </b>
       deposit
    </span>
    <br />
    <br />
    <button
      className="btn number cancel cancelBtn confirmation-cancel-btn"
      onClick={[Function]}
    >
      Cancel (ESC)
    </button>
    <button
      className="btn number success confirmBtn confirmation-ok-btn"
      onClick={[Function]}
    >
      OK (ENTER)
    </button>
  </div>
</div>
`;
